qutebrowser/browser/browsertab.py

```diff
@@ -515,7 +515,8 @@ def toggle_selection(self, line: bool = False) -> None:
     def drop_selection(self) -> None:
         raise NotImplementedError
 
-    def selection(self, callback: typing.Callable[[str], None]) -> None:
+    def selection(self, callback: typing.Callable[[str], None],
+                  rich: bool = False) -> None:
```

Comment:
```suggestion
                  *, rich: bool = False) -> None:
```

So that callers need to call `selection(cb, rich=True)` instead of `selection(cb, True)`.

Also, why `rich` and not `html`?

Reply:
Yes, `html` is a better name here. Btw for the key bindings, do you we should use `yh` or `hr` considering that the selected HTML will be converted to rich format by `QMimeData.setHtml`. And we now have to use `yy` instead of `y` to yank plain text. Do you think it will annoy existing users?

Reply:
Ah, I thought you could actually paste the clipboard content as plain text and get HTML back - but it looks like indeed it's pasted as rich text in e.g. Libreoffice.

I guess it's fine to call it "rich" from a user's perspective then! Still, in the tab API it should probably be called "html". As for the shortcut, I'd say `yr` then, as that's what it's doing when looking from the outside.

I'm fine with changing `y` to `yy` in caret mode.

Reply:
> but it looks like indeed it's pasted as rich text in e.g. Libreoffice.

You can "paste unformatted" in libreoffice which is a seperate action.

Reply:
> > but it looks like indeed it's pasted as rich text in e.g. Libreoffice.
>
> You can "paste unformatted" in libreoffice which is a seperate action.

Currently, this PR doesn't send formatted text to the clipboard. So "paste unformatted" in libreoffice will lead to empty content. I'm not sure if it's better to send both formatted text and plain text to the clipboard. I notice If we use mouse to copy selected content, we have both formattted and plain text. Does users also expect this behavior when they use the select&copy function in qutebrowser? @The-Compiler

Reply:
I don't really have an opinion on that. However, this discussion made me realize there might be a much easier implementation of copying possible. I'll comment on the PR instead of continuing inside this thread.